src/v/transform/api.cc

```diff
@@ -663,8 +668,10 @@ service::delete_transform(model::transform_name name) {
 
 ss::future<std::error_code> service::deploy_transform(
   model::transform_metadata meta, model::wasm_binary_iobuf binary) {
-    if (!_feature_table->local().is_active(
-          features::feature::wasm_transforms)) {
+    if (
+      !_feature_table->local().is_active(features::feature::wasm_transforms)
+      || !_feature_table->local().is_active(
+        features::feature::transforms_specify_offset)) {
```

Comment:
This blocks all deploys during the upgrade - we should only block usage of new position offsets because old versions will crash trying to read them.

Reply:
Right, there might be a disconnect in my understanding of how the variant serialization works. Won't we invariably write out a transform_offset_options which is versioned on the new variant layout?

Reply:
like if we allow a deploy (any deploy) to go through on the new version before the upgrade is complete, then we need to downgrade, won't metadata deser trigger this code?

https://github.com/redpanda-data/redpanda/blob/fd1434d925ceb5e50a67cd10d17043e5904b509b/src/v/serde/rw/variant.h#L146-L156

Reply:
done, i think. needs a test of some kind, but posting now for visibility